Generate a ChimeraX command file to open results in ChimeraX/ISOLDE

Guillawme opened the pull request:
Hi,

This PR adds code to generate a command file for ChimeraX to open the results of `refine_spa` in ChimeraX/ISOLDE. The new code mirrors closely what is done to generate the script for Coot, so I think it should work fine. But I could not test it because the build failed when I tried to `pip install` my local repository into a clean conda environment (I could check that this build failure is not caused by the new code, but otherwise I don't have time to investigate and fix it so I could test this new code).

Guillawme committed:
Generate a ChimeraX command file to open results in ChimeraX/ISOLDE

keitaroyam commented:
Thanks for the PR. I'm afraid I'm not using those programs so I cannot test them for now. I may come back later.

Guillawme commented:
If you can build the code at this commit and run a test job with it, then paste the content of the generated `{prefix}_chimerax.cxc`, I should be able to tell you if it works. I don't need to be able to run the commands in this cxc file to tell whether they are correctly written (I already tried that locally, which is how I came up with these commands, so I know they work), so I don't need the cif and mrc files. I only need to check that this cxc file is correctly generated when running a `servalcat refine_spa` job.

keitaroyam committed:
Update run_refmac.py

fix diffmap prefix

keitaroyam committed:
Update fofc.py

Fo-Fc +/- contouring. thanks https://obsidian-gallimimus-08c.notion.site/ChimeraX-256372918a0d4cf2822503134c9b3fcb

keitaroyam committed:
Update run_refmac.py

show cxc file name in refine_spa as well

keitaroyam commented:
Merged with some tweaks. Thank you

Guillawme commented:
Great, thank you!
Do you already know in which release will this become generally available?

Guillawme commented on servalcat/spa/fofc.py:
```diff
@@ -377,6 +377,17 @@ def write_coot_script(py_out, model_file, mtz_file, contour_fo=1.2, contour_fofc
+        ofs.write('open {}\n'.format(model_file))
+        ofs.write('open {}\n'.format(fo_mrc_file))
+        ofs.write('open {}\n'.format(fofc_mrc_file))
+        ofs.write('volume #3 level 4 level -4 color #00FF00 color #FF0000 squaremesh false cap false style mesh meshlighting false\n')
```
I think this line should be moved to the end, after both `clipper associate` commands. Because `clipper associate` will change the display style, so these customization will be lost if they run before `clipper associate`.

keitaroyam commented:
This change is included in 0.4.73. Probably I will release 0.4.74 or 75 soon, as there are some bug fixes for CCP4.

Guillawme commented:
Ok. I wanted to let you know that your addition of this line won't work, because the `clipper associate` commands that come after it will reset the display style.

```
volume #3 level 4 level -4 color #00FF00 color #FF0000 squaremesh false cap false style mesh meshlighting false
```

Even running this command last, after all the `clipper associate` commands, won't work as intended because clipper needs to be told that this is a difference map. This can be done in ISOLDE's graphical interface, but I could not find a documented command to do it from a script. See https://github.com/tristanic/chimerax-clipper/issues/4

I agree with you it's better to show the difference map with this style, but it doesn't seem doable from a command at the moment, due to the way clipper works.
